qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -197,16 +305,14 @@ def mcry(
                 target_qubit,
                 use_basis_gates=use_basis_gates,
             )
+        else:
+            mcsu2_real_diagonal(self, RYGate(theta).to_matrix(), control_qubits, target_qubit)
     else:
         raise QiskitError(f"Unrecognized mode for building MCRY circuit: {mode}.")
 
 
 def mcrz(
-    self,
-    lam: ParameterValueType,
-    q_controls: Union[QuantumRegister, List[Qubit]],
-    q_target: Qubit,
-    use_basis_gates: bool = False,
```

Comment:
We'll have to deprecate this behavior before removing it. If the new code uses different gates, one option could be to call `transpile` with `optimization_level=0` and `basis_gates=["p", "u", "cx"]` on the decomposition.

To deprecate this argument, you could check out this example: https://github.com/Qiskit/qiskit-terra/blob/a461a913c2e5bd51b307f38b89d2ab37117b7bf4/qiskit/algorithms/amplitude_amplifiers/grover.py#L117

Reply:
I added back the `use_basis_gates: bool = False`. Im not sure if this PR is the right place to include the deprecation note.